### PR TITLE
Initial Error Checking added

### PR DESCRIPTION
**Pull request by ryan280**

### CODE REVIEW

**Comment by jwolfcrafter** [comment]
You uploaded the node_modules folder...

**Comment by mpolimen** [comment]
I fixed a merge conflict with package.json and deleted package-lock.json.

**Comment by mpolimen** [comment]
Did not use JSDoc like we asked. I'll let it slide to make this get merged sooner. The rest seems good (although I can't personally vouch for the regex). Are we doing a story point limit of 5?

File: src/errors/error-handler.js [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+
+//These functions are meant to be used in if statements, so that you can make your own throw statement.
+
+function checkObjectId(OId){
+    if(ObjectId.isValid(OId)){
+        if((String)(new ObjectId(OId)) == OId){
+            return true;        
+        }
+        return false;
```
Comment (roccopolimen) [inline src/errors/error-handler.js:12]: if(ObjectId.isValid(OId))

     return (String)(new ObjectId(OId)) == OId;

return false;

// simplify the logic

File: src/errors/error-handler.js [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+//Checks if input is a valid hex color code
+function checkColor(colorCode){
+    return (typeof(colorCode) === 'string' && /^#([A-Fa-f0-9]{6}|[A-Fa-f0-9]{3})$/.test(colorCode));
+}
+
+//Checks if input is a valid array of objectid's
+function checkArrayObjectId(boardArray){
+    if(!Array.isArray(boardArray)){
+        console.log('here')
```
Comment (roccopolimen) [inline src/errors/error-handler.js:54]: remove the console.log()

File: src/errors/error-handler.js [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+        if(!checkObjectId(x)){
+            return false;
+        }
+    }
+    return true;
+}
+
+
+//Checks if input is a valid date in the form MM/DD/YYYY
+function checkDate(dateStr){
```
Comment (roccopolimen) [inline src/errors/error-handler.js:67]: Didn't this date code not work when he tested it? I have robust date code from a previous assignment we can use here

File: src/package.json [hunk 4]
```diff
@@ -18,6 +18,7 @@
   },
   "homepage": "https://github.com/roccopolimen/board-hub#readme",
   "dependencies": {
+    "xdate": "^0.8.2",
```
Comment (roccopolimen) [inline src/package.json:21]: won't be necessary with the other date checking code that uses native Date()

**Review by roccopolimen — COMMENTED** [review]
I left some comments in your code. We obviously need to talk about this a little bit as regex is kinda hard for me to read and verify, but thats what testing this weekend will help verify. Also not sure if we covered every error checker we will need, but once again, only time will tell and we can add things when we think of them. Generally this is a decent base, aside from the checkDate() method which I think needs to be changed.

**Comment by jwolfcrafter** [comment]
I also think there's a few methods we might want to add:
Check non-empty string (make sure its not a string filled with spaces)
Check positive number/check negative number
Just expansions of the other methods, you could even use the already written methods in these 3, they just feel useful.

**Comment by mpolimen** [comment]
Make names more inclusive

**Pull request merged**